streaming/rdm_atomic.c

```diff
@@ -170,63 +243,87 @@ static int run_op(void)
 	case FI_LXOR:
 	case FI_BXOR:
 	case FI_ATOMIC_WRITE:
-		for (datatype = 0; datatype < FI_DATATYPE_LAST; datatype++) {
-			ret = check_base_atomic_op(ep, op_type, datatype, count);
+		if (run_all_datatypes) {
+			for (datatype = 0; datatype < FI_DATATYPE_LAST; datatype++) {
+				ret = check_compare_atomic_op(ep, op_type,
+							      datatype, count);
+				if (ret == -FI_ENOSYS || ret == -FI_EOPNOTSUPP) {
+					fprintf(stderr,
+						"Provider doesn't support %s ",
+						fi_tostr(&op_type,
+							 FI_TYPE_ATOMIC_OP));
+					fprintf(stderr,
+						"compare atomic operation on %s\n",
+						fi_tostr(&datatype,
+							 FI_TYPE_ATOMIC_TYPE));
+					continue;
+				} else if (ret) {
+					break;
+				}
+
+				ret = execute_atomic_op(base, test_name, opts,
+							op_type, datatype);
+			}
+		} else {
+			ret = check_compare_atomic_op(ep, op_type,
+						      datatype, count);
 			if (ret == -FI_ENOSYS || ret == -FI_EOPNOTSUPP) {
-				fprintf(stderr, "Provider doesn't support %s ",
-					fi_tostr(&op_type, FI_TYPE_ATOMIC_OP));
-				fprintf(stderr, "base atomic operation on %s\n",
-					fi_tostr(&datatype, FI_TYPE_ATOMIC_TYPE));
-				continue;
```

Comment:
If the provider returns ENOSYS or EOPNOTSUPP, seems to me like we shouldn't test it, but I think this would test it anyway.

Reply:
Sorry, probably I got your comment wrong.
Seems it will report that provider doesn't support the datatype/operation and `continue` to the next datatype

Reply:
So if it's running all datatypes, it goes to the continue (line 259) which is correct. But if it's only running one datatype here and the provider says it does not support it, I think it should break or something instead of testing it anyway.

Reply:
Thank you. Got it :)